xla/service/gpu/gemm_rewriter.cc

```diff
@@ -176,13 +176,13 @@ bool IsSupportedF8Pattern(HloInstruction *instr, HloInstruction *&x,
   };
   for (int i = 3; i < subgraph->size(); ++i) {
     // The remaining instructions must be commutative with dequantization.
-    // Bitcast, broadcast, copy, pad, reshape and slice instructions are
-    // supported.
+    // Bitcast, broadcast, copy, pad, reshape, slice and all-gather instructions
+    // are supported.
     if (!Match((*subgraph)[i],
                m::AnyOf<HloInstruction>(
                    m::Bitcast().WithPredicate(preserves_element_type),
                    m::Broadcast(), m::Copy(), m::Pad(), m::Reshape(),
-                   m::Slice()))) {
+                   m::Slice(), m::AllGather()))) {
```

Comment:
@jurahul told me we should only change the dtype of all gather when SMPD is used and not MPMD, since in the MPMD case it's not guaranteed this optimization will be done for all replicas. You can check if SMPD is used with `module->config().use_spmd_partitioning()`

Reply:
Resolved.